app/services/notification_generator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Multiple children for expiring notification? One for each kid or a single one?
+class NotificationGenerator
+  def call
+    Notification.destroy_all
```

Comment:
How might we do this without needing to destroy all the notification records?  What errors could this cause?

As a thought experiment:
- what happens if the job fails in the middle and the Notifications have all been destroyed but the new ones haven't finished generating?

I might suggest a different implementation, like:

- create a variable of all approvals w/ a notification i.e. `approvals_with_notification = Approval.joins(:notification)`
- loop through each of those approvals
  - if the approval is still expiring within 30 days, `return` (do nothing)
  - if not, loop through all the notifications
    - if the notification's child has no approvals after the current approval, `return` (do nothing)
    - if they do have an approval afterwards, delete the notification

- create a variable of approvals expiring within 30 days w/ no notification i.e. `approvals_without_notification = Approval.where.missing(:notification).where('expires_on < ?', 30.days.after`
- if that variable length is greater than 0, loop through each and create a new notification.

Does this make sense?  Let me know what you think.